### PR TITLE
[system-probe] Fix concurrent map access

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Fix concurrent map access issue which was causing system-probe to crash when experimental HTTP monitoring feature was enabled. Since code is in hot code path I'm using atomic operations.

(Additionally I refactored the `Path()` function` to exclude GET variables, but this should have no side effect)

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

1. Deploy agent in load-test cluster
2. Make sure it doesn't crash after an hour or so
3. Verify that stats are showing up correctly in `curl -XGET --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/debug/stats | jq .network_tracer.http`

### CODE REVIEW

File: pkg/network/http/telemetry.go [hunk 1]
```diff
@@ -3,54 +3,56 @@
...
-	t.hits = make(map[int]int)
-	t.misses = 0
+	for i := range t.hits {
+		count := atomic.SwapInt64(&t.hits[i], 0)
+		data[fmt.Sprintf("%dXX_request_count", i+1)] = count
+		data[fmt.Sprintf("%dXX_request_rate", i+1)] = int64(float64(count) / elapsed)
+	}
+
+	data["requests_missed_count"] = misses
+	data["requests_missed_rate"] = int64(float64(misses) / elapsed)
```
Comment (brycekahle) [inline pkg/network/http/telemetry.go:55]: Why use `float64` if you then convert it back to `int64`? I think if `elapsed` is an `int64`, then all the casts goes away?

Comment (p-lambert) [reply pkg/network/http/telemetry.go:55]: huh I don't know, I was just copying/paste the old code but you're right. I'll fix it

File: pkg/network/http/model.go [hunk 2]
```diff
@@ -50,7 +51,7 @@ func (tx *httpTX) Path() string {
 
 	i++
 
-	for j = i; j < len(b) && b[j] != ' '; j++ {
+	for j = i; j < len(b) && b[j] != ' ' && b[j] != '?'; j++ {
```
Comment (ISauve) [inline pkg/network/http/model.go:54]: Nit: I think it's worth updating the comment above to indicate that this function removes query variables from the URL before returning it

**Review by ISauve — APPROVED** [review]
Nice!

File: pkg/network/http/model.go (hunk 2 above)
Comment (p-lambert) [reply pkg/network/http/model.go:54]: Good point, I'll update it 👍

**Pull request merged**